pkg/chunk/cache/fifo_cache.go

```diff
@@ -183,91 +213,116 @@ func (c *FifoCache) Put(ctx context.Context, keys []string, values []interface{}
 	}
 }
 
-func (c *FifoCache) put(ctx context.Context, key string, value interface{}) {
-	// See if we already have the entry
-	index, ok := c.index[key]
-	if ok {
-		entry := c.entries[index]
-		deltaSize := sizeOf(value) - sizeOf(entry.value)
+func (c *FifoCache) addToHead(entry *cacheEntry) {
+	if entry == nil {
+		return
+	}
```

Comment:
Remove this check please. If entry is `nil`, let it crash so that we fix it quickly. It should never be called with `nil` pointer.